src/main/java/Cars.java

```diff
@@ -12,9 +12,9 @@ public List<Car> getValue() {
         return this.value;
     }
 
-    public void move() {
+    public void move(MovePolicy movePolicy) {
         for (Car car : this.value) {
-            car.move();
+            car.move(movePolicy.isAbleToMove());
```

Comment:
```suggestion
            car.move(movePolicy);
```
movePolicy를 car.move()에 전달하고 move() 메서드가 호출하도록 구현하는 것은 어떨까?

Reply:
네, car가 move하기위한 조건(policy)을 전달해주는 것이 좀 더 객체지향적인 설계로 보이네요. 적용해보았습니다 :)